compiler/src/iree/compiler/Codegen/Common/FissionTransferOpsInControlFlow.cpp

```diff
@@ -119,7 +119,8 @@ static void setupReadLoop(IRRewriter &rewriter, vector::TransferReadOp readOp,
 
   auto readLoopIndex = createMemrefAccessIndex(rewriter, readLoop);
   SmallVector<Value> readLoopIndices = {readLoopIndex};
-  auto readOpIndicesSize = readOp.getIndices().size();
+  // Use alloca size to determine the number of zero indices needed.
+  auto readOpIndicesSize = alloca.getType().getShape().size() - 1;
```

Comment:
This change seems to have snuck in and is unrelated?

Reply:
Add a test for that with this PR?

Reply:
Good point! let me add one